### PR TITLE
navbar in vue app

### LINKED ISSUE #22: Navbar in vue app
# I want to be able to go through different routes using navbar for better user experience.

## Acceptance criteria:
The navbar is responsive.
The navbar contains all routes and user profile information.

### PR DESCRIPTION
**Pull request by AfaqAhmed-XGrid**
## **Issue** #22   
     Title: Navbar in vue app

## Description
This pull request sets up the navbar in vue app:
=> Navbar: Used quasar and custom scss to design navbar that is fully responsive.
=> Http request: Used axios to make http request to get user data.
=> Routes Protection: Added routes protection
=> Scroll up button: Added scroll up button

## **Checklist**
- [x] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the README.md, or the docs directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes (if applicable)
- [x] Every function, interface, class has a comment describing what it does and input/output parameters

### CODE REVIEW

File: ui_vue/src/App.vue [hunk 1]
```diff
@@ -44,14 +47,54 @@ watch(route, () => {
...
+            <q-btn fab icon="keyboard_arrow_up" color="accent" padding="md"/>
           </q-page-scroller>
         </q-page>
       </q-page-container>
     </q-layout>
   </div>
 </template>
+
+<style scoped lang="scss">
+@import './styles/breakpoints.scss';
```
Comment (XGMqasim) [inline ui_vue/src/App.vue:74]: Why the styles are not in a separate file?

File: ui_vue/src/components/TheNavBar.vue [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+    v-model="drawer"
+    show-if-above
+    :width="200"
+    :breakpoint="500"
+    elevated
+    class="drawer-bg"
+  >
+    <q-scroll-area class="fit">
+      <q-list padding>
+        <q-item clickable v-ripple :active="route.path === '/dashboard'" to="/dashboard">
```
Comment (XGMqasim) [inline ui_vue/src/components/TheNavBar.vue]: You can loop through all these elements

File: ui_vue/src/composables/useAuthHttpRequests.ts [hunk 3]
```diff
@@ -135,5 +135,31 @@ export const useAuthHttpRequest = () => {
...
-  return { signInUser, signUpUser, authCallBack, resetUserPassword };
+  /**
+   * Function to make http request to sign in the user with credentials
+   * @return {Promise<GetUserSuccessResponse | void>} returns a promise
+   */
+  const getUser = async (): Promise<User | void> => {
+    try {
+      const token = localStorage.getItem(constants.localStorage.userToken);
+      const response = await axios.get(baseURL + authEndPoint.getUser, {
+        headers: {
```
Comment (XGMqasim) [inline ui_vue/src/composables/useAuthHttpRequests.ts]: You can use axios interceptors for this otherwise you have to add headers in each request

**Review by XGMqasim — CHANGES_REQUESTED** [review]
-1: Address all comments

File: ui_vue/src/App.vue (hunk 1 above)
Comment (AfaqAhmed-XGrid) [reply ui_vue/src/App.vue:74]: moved to the dedicated styling files

File: ui_vue/src/components/TheNavBar.vue (hunk 2 above)
Comment (AfaqAhmed-XGrid) [reply ui_vue/src/components/TheNavBar.vue]: Done

File: ui_vue/src/composables/useAuthHttpRequests.ts (hunk 3 above)
Comment (AfaqAhmed-XGrid) [reply ui_vue/src/composables/useAuthHttpRequests.ts]: Done

**Review by XGMqasim — APPROVED** [review]

**Review by saif-xgrid — APPROVED** [review]
LGTM

**Pull request merged**